src/main/java/no/unit/nva/database/DatabaseServiceImpl.java

```diff
@@ -58,31 +66,53 @@ public DatabaseServiceImpl(DynamoDBMapper mapper) {
         return Optional.ofNullable(user);
     }
 
-    private DynamoDBQueryExpression<UserDb> createGetUserQuery(UserDb searchObject) {
-        return new DynamoDBQueryExpression<UserDb>()
+    @Override
+    public void addUser(UserDto user) throws InvalidUserException {
+        mapper.save(user.toUserDb());
+    }
+
+    @Override
+    public void addRole(RoleDto roleDto) throws InvalidInputRoleException {
+        Try.of(roleDto)
+            .forEach(role -> mapper.save(roleDto.toRoleDb()))
+            .orElseThrow(failure -> new InvalidInputRoleException(INVALID_ROLE_ERROR));
+    }
+
+    @Override
+    public UserDto updateUser(UserDto user) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    @Override
+    public Optional<RoleDto> getRole(RoleDto input) throws InvalidRoleException {
```

Comment:
I am just mimicking the DynamoDB approach. This gives a minimalist DatabaseService actually. See it as query by example, the input is an object that defines the desired values on specific attributes. So instead of having `getByRoleName`, `getByAttributeAandAttributeB()` you have only one method.